framework/src/wifi_manager/wifi_manager_lwnl_listener.c

```diff
@@ -143,10 +148,11 @@ int lwnl_fetch_event(int fd, void *buf, int buflen)
 {
 	lwnl_cb_status status;
 	uint32_t len;
-	char type_buf[8] = {0,};
+	char type_buf[sizeof(lwnl_cb_status) + sizeof(uint32_t)] = {0,};
+	size_t buf_len = sizeof(lwnl_cb_status) + sizeof(uint32_t);
 	handler_msg *hmsg = (handler_msg *)buf;
 
-	int nbytes = read(fd, (char *)type_buf, 8);
+	int nbytes = read(fd, (char *)type_buf, buf_len);
 	if (nbytes < 0) {
```

Comment:
It is error if only nbytes is negative integer with current condition, `if (nybytes < 0)`.
A value of nbytes can be 0 or positive integer not buf_len(8) like 3, 5.
It is invalid but not handled here and can make side effect later.
I think it should read all 8 bytes to succeed in this case.
So how about changing it to `if (nbytes != buf_len)`?

Reply:
file descriptor type is lwnl. if succeeds to read then it guarantees the read size. so it doesn't need to consider above case.